### PR TITLE
Fix some wording on the example page regarding the .tcd and .tdr files

### PR DESCRIPTION
**Pull request by jainam1995**
Issue : https://github.com/tableau/connector-plugin-sdk/issues/518 .
Tried rewording the statements to keep the intent but make it less confusing. Open to better re structuring of the statements.

### CODE REVIEW

File: docs/example.md [hunk 1]
```diff
@@ -6,8 +6,8 @@ A Tableau connector is a set of files that describe the UI elements needed to co
...
-- __Tableau Custom Dialog file__ customizes the connection dialog as needed for this connection. (If the base is not set to "odbc" or "jdbc", then this is file may not be required.)
-- __Tableau Connection Resolver file__ creates the connection to your file using JavaScript files to define and make the connection. (If the base is not set to "odbc" or "jdbc", then this is file may not be required.)
+- __Tableau Custom Dialog file__ customizes the connection dialog as needed for this connection. This file is optional. If not created, it will be based off the superclass set in the Manifest file (manifest.xml) by default. However if the superclass is set to "odbc" or "jdbc", then this file is required.
+- __Tableau Connection Resolver file__ creates the connection to your file using JavaScript files to define and make the connection. This file is optional. If not created, it will be based off the superclass set in the Manifest file (manifest.xml) by default. However if the superclass is set to "odbc" or "jdbc", then this file is required.
```
Comment (pvanderknyff) [inline docs/example.md:10]: I think this is written backwards. 99% of people should be using odbc or jdbc, so in 99% of the people reading this will be required to have these files. However, as it stands the first thing they read is "This file is optional," full stop, and they may not read further. 

I actually think the original is clearer on this - this file _may not_ be required in certain, very specific situations. I would hesitate to change that narrative.

**Comment by pvanderknyff** [comment]
The original issue is about people mistaking the line to mean they can do something that isn't odbc or jdbc based, this leaves that ambiguity.

**Comment by jainam1995** [comment]
> The original issue is about people mistaking the line to mean they can do something that isn't odbc or jdbc based, this leaves that ambiguity.

Yes, the options other than odbc or jdbc are the classes they mention in the superclass . If these are not odbc and jdbc , in that case this file becomes optional but if they mention it to be odbc or jdbc then this file is required. That's what i tried to point out by mentioning the superclass information in the change. 

I will rephrase the change to not change the "may to be required".

File: docs/example.md (hunk 1 above)
Comment (jainam1995) [reply docs/example.md:10]: I will rephrase it to keep the original narrative,

**Comment by jainam1995** [comment]
> The original issue is about people mistaking the line to mean they can do something that isn't odbc or jdbc based, this leaves that ambiguity.

Does adding information about the usperclass which is the throd and only option after odbc and jdbc help in the direction of solving the ambiguity ?

File: docs/example.md (hunk 1 above)
Comment (jainam1995) [reply docs/example.md:10]: Was thinking of something like  (If the superclass in Manifest file (manifest.xml) is not set to the value “odbc” or “jdbc”, then this is file may not be required and it will be based off the odbc/jdbc based superclass specified.

File: docs/example.md [hunk 2]
```diff
@@ -6,8 +6,8 @@ A Tableau connector is a set of files that describe the UI elements needed to co
 
 Tableau connector files include:
 - __Manifest file__ that tells Tableau about the connector. Tableau uses the name from this file to add it to the list of available connectors in the UI.
-- __Tableau Custom Dialog file__ customizes the connection dialog as needed for this connection. (If the base is not set to "odbc" or "jdbc", then this is file may not be required.)
-- __Tableau Connection Resolver file__ creates the connection to your file using JavaScript files to define and make the connection. (If the base is not set to "odbc" or "jdbc", then this is file may not be required.)
+- __Tableau Custom Dialog file__ customizes the connection dialog as needed for this connection. This file is optional. If not created, it will be based off the superclass set in the Manifest file (manifest.xml) by default. However if the superclass is set to "odbc" or "jdbc", then this file is required.
```
Comment (pvanderknyff) [inline docs/example.md:9]:
```suggestion
- __Tableau Custom Dialog file__ customizes the connection dialog as needed for this connection. In most cases, this is required. However, if you subclass from an existing connector like MySQL or Postgres, and you do not create this file, we will use the default for that connector.
```

Comment (pvanderknyff) [inline docs/example.md:9]:
```suggestion
- __Tableau Custom Dialog file__ customizes the connection dialog as needed for this connection.
```

Comment (pvanderknyff) [reply docs/example.md:9]: This would be my preferred option, since it's unlikely and discouraged for them to a) use an existing connector as a superclass and b) if they do use an existing connector, use the base files.

Comment (jainam1995) [reply docs/example.md]: This is what i prefer too after looking at the confusion the term optional is creating. I made the necessary changes,
